mtls_server/auth.py

```diff
@@ -0,0 +1,93 @@
+import base64
+import os
+import time
+import pprint
+import json
+from functools import wraps
+
+from flask import request
+from flask import g
+from flask import current_app
+
+from .logger import logger
+from .utils import error_response
+from .utils import time_in_range
+from .utils import write_sig_to_file
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+
+class SignatureTimestampOutOfBoundsException(Exception):
+    pass
+
+
+class MissingTokenException(Exception):
+    pass
+
+
+class BadSignatureException(Exception):
+    pass
+
+
+def login_required(f):
+    @wraps(f)
+    def login_required_wrap(*args, **kwargs):
+        allowed_tokens = ['PGP-SIG']
+        g.is_admin = False
+
+        if not request.headers['Authorization']:
+            return error_response("authentication required", 401)
+
+        token_type, token = request.headers['Authorization'].split(' ')
```

Comment:
Might be worth using the walrus here?
```suggestion
        if not (authorization_header := request.headers['Authorization']):
            return error_response("authentication required", 401)

        token_type, token = authorization_header.split(' ')
```